File: frontend/src/types/analytics.ts
export interface AnalyticsSummary {
  total_requests: number;
  sent_requests: number;
  opened_requests: number;
  clicked_requests: number;
  completed_requests: number;
  conversion_rate: number;
}

export interface DailyTrend {
  date: string;
  count: number;
}

export interface AnalyticsDashboard {
  summary: AnalyticsSummary;
  daily_trends: DailyTrend[];
}

export interface ActivityItem {
  id: string;
  type: 'sent' | 'opened' | 'clicked' | 'completed';
  customerName: string;
  timestamp: string;
}

File: backend/app/schemas/analytics.py
from typing import List, Dict
from pydantic import BaseModel
from datetime import date

class AnalyticsSummary(BaseModel):
    total_requests: int
    sent_requests: int
    opened_requests: int
    clicked_requests: int
    completed_requests: int
    conversion_rate: float

class DailyTrend(BaseModel):
    date: date
    count: int

class AnalyticsDashboard(BaseModel):
    summary: AnalyticsSummary
    daily_trends: List[DailyTrend]
